### ISSUE TITLE
Bring CPR fork changes upstream

### ISSUE THREAD

**Issue by efernandez** [ros-controls/ros_controllers#295]
Bring changes since June 8, 2015.

- [ ] First PR is in, waiting on @bmagyar 's review: https://github.com/ros-controls/ros_controllers/pull/294

- [ ] Next PR should take the 5 commits from https://github.com/clearpathrobotics/ros_controllers/pull/3/commits, which will bring the following features:

1. Separate wheel radii
2. Covariance computation
3. Roslinting and warning fixes
4. New unit tests (mainly for the new covariance computation functionality)

- [ ] TBD (I'd update this as I create new PR and we go forward, since all commits are on top of each other, and things need to be reviewed and merged sequentially).

FYI @bmagyar @artivis
